benchmarks/bench_text_vectorizers.py

```diff
@@ -0,0 +1,73 @@
+"""
+
+To run this benchmark, you will need,
+
+ * scikit-learn
+ * pandas
+ * memory_profiler
+ * psutil (optional, but recommended)
+
+"""
+
+from __future__ import print_function
+
+import timeit
+import itertools
+
+import numpy as np
+import pandas as pd
+from memory_profiler import memory_usage
+
+from sklearn.datasets import fetch_20newsgroups
+from sklearn.feature_extraction.text import (CountVectorizer, TfidfVectorizer,
+                                             HashingVectorizer)
+
+
+def run_vectorizer(Vectorizer, X, **params):
+    def f():
+        vect = Vectorizer(**params)
+        vect.fit_transform(X)
+    return f
+
+
+text = fetch_20newsgroups(subset='train').data
+
+print("="*80 + '\n#' + "    Text vectorizers benchmark" + '\n' + '='*80 + '\n')
+print("Using a subset of the 20 newsrgoups dataset ({} documents)."
+      .format(len(text)))
+print("This benchmarks runs in ~20 min ...")
+
+res = []
+
+# Wrap the result of iteritools.product with tqdm to get a progress bar
+for Vectorizer, (analyzer, ngram_range) in itertools.product(
+            [CountVectorizer, TfidfVectorizer, HashingVectorizer],
+            [('word', (1, 1)),
+             ('word', (1, 2)),
+             ('word', (1, 4)),
+             ('char', (4, 4)),
+             ('char_wb', (4, 4))
+             ]):
+
+    bench = {'vectorizer': Vectorizer.__name__}
+    params = {'analyzer': analyzer, 'ngram_range': ngram_range}
+    bench.update(params)
+    dt = timeit.repeat(run_vectorizer(Vectorizer, text, **params),
+                       number=1,
+                       repeat=3)
+    bench['time'] = "{:.2f} (±{:.2f})".format(np.mean(dt), np.std(dt))
```

Comment:
Thanks, forgot about this. Added an an encoding declaration on the top. 

I agree that computing it on 3 points is non ideal, but these benchmarks already take quite some time (20min) and I felt that the current 10k document dataset would be quite representative of a real work dataset (vocabulary size wise etc), and and wasn't sure about reducing it's size to make the computations faster. Besides, in this benchmark the std is ~1% or the run time, so even if it's off by a factor of 2 or so, it doesn't really matter for any optimization that gains more than 5% performance. It was about giving an idea of the order magnitude of the standard deviation, to say that run to run variations are almost negligible.

> If so, it might be better to show the median and maybe range, not sure...

Well the thing is that when running this benchmarks on multiple configurations these tables are already hard to parse as it is (e.g. see [this example](https://github.com/rth/notebooks/blob/master/benchmark/benchmark_count_vectorizer.md) ) adding a range would be even less readable, I think. About median, you might be right but the mean is more consistent with the regular `%timeit` typically used in benchmarks and should be more stable run to run..

What do you think? Also thanks for the review! Removed the the comment over `iteritools.product`, you are right.